RFS/src/main/java/org/opensearch/migrations/parsing/BulkResponseParser.java

```diff
@@ -0,0 +1,120 @@
+package org.opensearch.migrations.parsing;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonToken;
+
+import lombok.Builder;
+import lombok.Getter;
+import lombok.experimental.UtilityClass;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@UtilityClass
+public class BulkResponseParser {
```

Comment:
Is there a speed or efficiency gain by parsing the JSON like this at a lower level rather than marshalling the response into objects?  If there isn't a big difference, it feels like it would be simpler to just do the marshalling and scan through that way.  We're taking a dependency on the version-specific structure of the response body either way.

Reply:
No matter the speed difference, if we get a truncated response for whatever reason this parser will still be able to read the json document until it hits the end of what is available.  Added tests that prove this.

Additionally, this parser should be able to handle streaming input with minimal changes if/when that arrives.

Finally its ~19% faster than using readObject, I'll create a separate PR with the details.

```
Benchmark                                                                      Mode  Cnt   Score   Error  Units
JacksonParserVsReadObjectBenchmark.testJacksonMarshalling                     thrpt    2  56.948          ops/s
JacksonParserVsReadObjectBenchmark.testJacksonParser                          thrpt    2  67.714          ops/s
```